cdn/src/handler.ts

```diff
@@ -19,8 +19,15 @@ type CLIArgs = {
 
 type PossiblePaths = [undefined, Product?, CLIArgs["platform"]?, string?];
 
+const SUPPORTED_METHODS = ["GET", "HEAD"]
 export async function handleRequest(event: FetchEvent): Promise<Response> {
   try {
+    if (!SUPPORTED_METHODS.includes(event.request.method)) {
+      return Promise.resolve(
+        new Response(`This proxy only supports the following request types: ${SUPPORTED_METHODS.join(", ")}`, { status: 500 })
+      )
```

Comment:
Is the wrapping in `Promise.resolve` necessary since we're already an `async` function that will return a `Promise`?